### ISSUE TITLE
Command And Control API : command TTL

### ISSUE THREAD

**Issue by ppatierno** [eclipse/hono#17]
In the command and control scenario, it's useful to specify the TTL (Time To Live) for a command.
Sometimes if the command isn't executed in a short time by the device (i.e. in 5 secs, 10 secs, 1 min, ...) because it's offline, it could be useless (or dangerous !!!) to execute it when the device will come back online after a long time (i.e. 30 min, 1 hour, ...).
For this reason specifying a TTL inside the command could be very useful.
It could be provided by the "ttl" field inside the Header section as defined in the AMQP specification.

Of course it's true if a "store and forward" approach is used where the command is delivered to the device using a queue for example. When a "direct" connection is setup between sender (the service) and receiver (the device) through a "dispatch router" for example, the TTL can't be used because the message isn't delivered to destination (the device is offline) but the dispatch router doesn't store it.
In that case the sender immediately knows that receiver isn't available thanks to the dispatch router behavior (disposition message with MODIFIED delivery state).

In conclusion, it's useful to add the TTL inside the message format table.

**Comment by sophokles73** [comment]
I agree, maybe @hekonsek wants to add it?

**Comment by grs** [comment]
On a related point, using the durable field in the message header would also be useful as a way of indicating whether the command should be stored if the intended target is not currently connected.

**Comment by ppatierno** [comment]
Good point @grs ! I propose to @sophokles73 and @hekonsek to add both ttl and durable.

**Comment by hekonsek** [comment]
Hi,

I have added both `ttl` and `durable` properties to the API specification. I also assumed that default TTL is one minute, and that command&control messages are durable by default.
